lucetc/src/decls.rs

```diff
@@ -528,7 +521,10 @@ impl<'a> ModuleDecls<'a> {
             let name = self
                 .function_names
                 .get(fn_index)
-                .ok_or_else(|| format_err!("func index out of bounds: {:?}", fn_index))
+                .ok_or_else(|| {
+                    let message = format!("{:?}", fn_index);
+                    Error::FunctionIndexError(message)
+                })
```

Comment:
This is trying to handle an error that should be impossible - fn_index should always be valid, so I'd just change it to `let name = self.function_names.get(fn_index).expect("fn_index is key into function_names");`. Thats the only other use of the FunctionIndexError variant, so you can delete it after this is gone

Reply:
Thanks!